src/modules/expenses/expense.repository.ts

```diff
@@ -3,47 +3,14 @@ import type { TInsertExpenseSchema } from "@/db/schemas/expense.model";
 import { db } from "@/db/adapter";
 import expenseModel from "@/db/schemas/expense.model";
 
-interface TExpensePayloadUser {
-  amount: TInsertExpenseSchema["amount"];
-  currency: TInsertExpenseSchema["currency"];
-  splitType: TInsertExpenseSchema["splitType"];
-  description?: TInsertExpenseSchema["description"];
-}
-
-interface TExpensePayloadAdmin {
-  amount: TInsertExpenseSchema["amount"];
-  currency: TInsertExpenseSchema["currency"];
-  splitType: TInsertExpenseSchema["splitType"];
-  payerId: TInsertExpenseSchema["payerId"];
-  description?: TInsertExpenseSchema["description"];
-}
-
-export async function createExpenseRepositoryViaUser(
-  expensePayload: TExpensePayloadUser,
-  userId: string,
-) {
-  const [expense] = await db
-    .insert(expenseModel)
-    .values({
-      ...expensePayload,
-      payerId: userId,
-      creatorId: userId,
-    })
-    .returning();
-
-  return expense;
-}
+type TExpensePayload = Pick<TInsertExpenseSchema, "amount" | "currency" | "splitType" | "creatorId" | "payerId" | "description" >;
```

Comment:
Suggestion: Just like Pick we also have Omit, whatever is less you can use that.

e.g: If you have 5 field and you want 4, using Omit is better and if the requirement is vice-versa using Pick is better.

Reply:
Cool. Done.